laa_crime_application_store_app/services/permissions.py

```diff
@@ -0,0 +1,47 @@
+from fastapi import Depends
+from fastapi_azure_auth.exceptions import InvalidAuth
+from fastapi_azure_auth.user import User
+from sentry_sdk import capture_message
+
+from laa_crime_application_store_app.config.permission_settings import get_permissions
+from laa_crime_application_store_app.models.application_schema import Application
+from laa_crime_application_store_app.services.auth_service import azure_schema
+
+permissins = get_permissions()
+
+
+def validate_can_create(user: User = Depends(azure_schema)) -> None:
+    if user is None or not user.roles:
+        capture_message("No roles setup for permission logic")
+        # remove this branch once roles have been implemented
+        # on Azure accounts
+    elif permissins.provider_role not in user.roles:
+        raise InvalidAuth("User not permitted to create application")
+
+
+def validate_can_update(
+    application: Application, user: User = Depends(azure_schema)
+) -> None:
+    if application.application_state in permissins.locked:
+        raise InvalidAuth("Application in locked state")
+    elif not user.roles:
```

Comment:
won't this fail if you have authentication turned off?

Reply:
Also if you wanted to we can use a match statement here as allowed from python 3.10 :)
https://docs.python.org/3/whatsnew/3.10.html#pep-634-structural-pattern-matching